plugins/aws-proxy-filebased-credentials-provider/src/main/java/io/trino/aws/proxy/plugins/filebasedcredentialsprovider/FileBasedCredentialsProvider.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.plugins.filebasedcredentialsprovider;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Files;
+import com.google.inject.Inject;
+import io.airlift.json.JsonCodec;
+import io.trino.aws.proxy.spi.credentials.Credential;
+import io.trino.aws.proxy.spi.credentials.Credentials;
+import io.trino.aws.proxy.spi.credentials.CredentialsProvider;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class FileBasedCredentialsProvider
+        implements CredentialsProvider
+{
+    private final Map<String, Credentials> credentialsStore;
+
+    @Inject
+    public FileBasedCredentialsProvider(FileBasedCredentialsProviderConfig config, JsonCodec<List<FileBasedCredentials>> fileBasedCredentialsCodec)
+    {
+        requireNonNull(config, "Config is null");
+        requireNonNull(fileBasedCredentialsCodec, "file based credentials codec is null");
+        this.credentialsStore = buildCredentialsMap(config.getCredentialsFile(), fileBasedCredentialsCodec);
+    }
+
+    private Map<String, Credentials> buildCredentialsMap(Optional<File> credentialsFile, JsonCodec<List<FileBasedCredentials>> fileBasedCredentialsCodec)
+    {
+        ImmutableMap.Builder<String, Credentials> credentialsMapBuilder = ImmutableMap.builder();
+        List<FileBasedCredentials> credentialsList;
+        try {
+            credentialsList = fileBasedCredentialsCodec.fromJson(Files.toByteArray(credentialsFile.orElseThrow()));
+        }
+        catch (IOException e) {
+            throw new RuntimeException("Failed to read credentials file", e);
+        }
+        for (FileBasedCredentials item : credentialsList) {
```

Comment:
Why not make the file have a complete JSON list and this won't be necessary?

Reply:
Well, the JSON has to be parsed and the Credentials provider has to return the credentials. I have updated the PR description with some more explanation for the usecase.

Reply:
See above. I have an example of that uses `Credentials` with an `ObjectMapper`